tests/search/search-test.spec.ts

```diff
@@ -0,0 +1,210 @@
+    expect(await sortDirectionSelector.count()).toEqual(1);
+    expect(await srOnlySortDirection.count()).toEqual(1);
+
+    expect(await sortByText.innerText()).toEqual('Sort by:');
+    expect(await srOnlySortDirection.innerText()).toEqual('Change to ascending sort');
+
+    const sortSelectorContainer = sortBarSection.locator('div#sort-selector-container');
+    const mobileSort = sortSelectorContainer.locator('#mobile-sort-container');
+    const desktopSort = sortSelectorContainer.locator('#desktop-sort-container');
+    
+    const desktopSortSelector = desktopSort.locator('#desktop-sort-selector');
+    const desktopSortSelectorTexts = await desktopSort.locator('ul#desktop-sort-selector').allInnerTexts();
+    const desktopSortTexts = Object.assign([], desktopSortSelectorTexts[0].split('\n'));
+
+    await desktopSortSelector.click();
+    await expect(page).toHaveURL(/sort=title/);
+
+    await page.waitForTimeout(5000);
+    expect(await srOnlySortDirection.innerText()).toEqual('Change to descending sort');
+
+    desktopSortTexts.forEach((text: string, ix) => {
+      expect(text.includes(sortTextList[ix]));
+    });
+    
+    expect(await sortSelectorContainer.count()).toEqual(1);
+    expect(await desktopSortSelector.count()).toEqual(1);
+
+    expect(await expect(mobileSort).toHaveClass(/hidden/));
+    expect(await expect(desktopSort).toHaveClass(/visible/));
+
+    // sort-bar-filter URL changes
+    await page.getByRole('button', { name: 'Relevance' }).click();
+
+    await page.getByText('Weekly views').first().click();
+    await expect(page).toHaveURL(/query=cats&sort=-week/);
+
+    await page.getByRole('button', { name: 'Toggle options Weekly views' }).getByRole('button').click();
+    await page.getByRole('button', { name: 'All-time views' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=-downloads/);
+
+    await page.getByRole('button', { name: 'Title' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=title/);
+
+    await page.getByText('Date published').first().click();
+    await expect(page).toHaveURL(/query=cats&sort=-date/);
+
+    await page.getByRole('button', { name: 'Toggle options Date published' }).getByRole('button').click();
+    await page.getByRole('button', { name: 'Date archived' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=-publicdate/);
+ 
+    await page.getByRole('button', { name: 'Toggle options Date archived' }).getByRole('button').click();
+    await page.getByRole('button', { name: 'Date reviewed' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=-reviewdate/);
+
+    await page.getByRole('button', { name: 'Toggle options Date reviewed' }).getByRole('button').click();
+    await page.getByRole('button', { name: 'Date added' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=-addeddate/);
+
+    await page.getByRole('button', { name: 'Creator' }).click();
+    await expect(page).toHaveURL(/query=cats&sort=creator/);
```

Comment:
Q: do we need to test result set here? to confirm that they are truly being sorted?